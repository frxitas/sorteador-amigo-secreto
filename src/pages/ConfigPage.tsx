import { DefaultLayout } from "../Layout/DefaultLayout";

import { Footer } from "../components/Footer/Footer";
import { Form } from "../components/Form/Form";
import { ParticipantList } from "../components/ParticipantList/ParticipantList";

export const ConfigPage = () => {
	return (
		<DefaultLayout>
			<section>
				<h3>Vamos come√ßar!</h3>
				<Form />
				<ParticipantList />
				<Footer />
			</section>
		</DefaultLayout>
	);
};
