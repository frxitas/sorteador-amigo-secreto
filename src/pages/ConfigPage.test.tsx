import React from "react";
import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import { ConfigPage } from "./ConfigPage";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
	return {
		useNavigate: () => mockNavigate,
	};
});

describe("config page", () => {
	test("correctly render", () => {
		const { container } = render(
			<RecoilRoot>
				<ConfigPage />
			</RecoilRoot>,
		);

    expect(container).toMatchSnapshot();
	});
});
