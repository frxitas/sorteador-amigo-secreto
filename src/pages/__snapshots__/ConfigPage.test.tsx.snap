// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config page correctly render 1`] = `
<div>
  <section
    class="css-12tmq8n"
  >
    <header
      class="css-j1px08"
    >
      Sorteador de Amigo Secreto
    </header>
    <div
      class="css-o7m9iu"
    >
      <section>
        <h3>
          Vamos come√ßar!
        </h3>
        <form>
          <div
            class="css-831njy"
          >
            <input
              class="css-1c8qenx"
              placeholder="Insira o nome do participante"
              type="text"
              value=""
            />
            <button
              class="css-ovsbhy"
              disabled=""
            >
              Adicionar
            </button>
          </div>
          
        </form>
        <div
          class="css-y52p8w"
        >
          <ul
            class="css-1ro3o6z"
          />
        </div>
        <footer
          class="css-1j2lcdp"
        >
          <button
            class="css-6wnd70"
            disabled=""
          >
            Iniciar
          </button>
        </footer>
      </section>
    </div>
  </section>
</div>
`;
